modules/storage-api/src/test/java/org/apache/ignite/internal/storage/util/MvPartitionStoragesTest.java

```diff
@@ -0,0 +1,606 @@
+
+        // What if there is an error during the operation?
+
+        assertThat(mvPartitionStorages.create(2, partId -> {
+            throw new RuntimeException("from test");
+        }), willFailFast(RuntimeException.class));
+
+        assertNull(getMvStorage(2));
+    }
+
+    @Test
+    void testCreateDuringDestroy() {
+        assertThat(createMvStorage(0), willCompleteSuccessfully());
+
+        MvPartitionStorage mvStorage = getMvStorage(0);
+
+        CompletableFuture<Void> startDestroyMvStorageFuture = new CompletableFuture<>();
+        CompletableFuture<Void> finishDestroyMvStorageFuture = new CompletableFuture<>();
+
+        CompletableFuture<?> destroyMvStorageFuture = runAsync(() ->
+                assertThat(mvPartitionStorages.destroy(0, mvStorage1 -> {
+                    startDestroyMvStorageFuture.complete(null);
+
+                    return finishDestroyMvStorageFuture;
+                }), willCompleteSuccessfully())
+        );
+
+        assertThat(startDestroyMvStorageFuture, willCompleteSuccessfully());
+
+        CompletableFuture<MvPartitionStorage> reCreateMvStorageFuture = createMvStorage(0);
+
+        assertThrowsWithMessage(StorageException.class, () -> createMvStorage(0),
+                "Creation of the storage after its destruction is already planned"
+        );
+
+        finishDestroyMvStorageFuture.complete(null);
+
+        assertThat(destroyMvStorageFuture, willCompleteSuccessfully());
+
+        assertThat(reCreateMvStorageFuture, willCompleteSuccessfully());
+
+        assertNotSame(mvStorage, getMvStorage(0));
+    }
+
+    @Test
+    void testCreateDuringDestroyError() {
+        assertThat(createMvStorage(0), willCompleteSuccessfully());
+
+        // What if there is an error during the destroy?
+
+        MvPartitionStorage mvStorage = getMvStorage(0);
+
+        CompletableFuture<Void> startErrorDestroyMvStorageFuture = new CompletableFuture<>();
+        CompletableFuture<Void> finishErrorDestroyMvStorageFuture = new CompletableFuture<>();
+
+        CompletableFuture<?> errorDestroyMvStorageFuture = runAsync(() ->
+                assertThat(mvPartitionStorages.destroy(0, mvStorage1 -> {
+                    startErrorDestroyMvStorageFuture.complete(null);
+
+                    return finishErrorDestroyMvStorageFuture;
+                }), willCompleteSuccessfully())
+        );
+
+        assertThat(startErrorDestroyMvStorageFuture, willCompleteSuccessfully());
+
+        CompletableFuture<MvPartitionStorage> errorReCreateMvStorageFuture = createMvStorage(0);
+
+        finishErrorDestroyMvStorageFuture.completeExceptionally(new RuntimeException("from test"));
+
+        assertThat(errorDestroyMvStorageFuture, willFailFast(RuntimeException.class));
+        assertThat(errorReCreateMvStorageFuture, willFailFast(RuntimeException.class));
+
+        assertNull(getMvStorage(0));
+    }
+
+    @Test
+    void testCreateAfterDestroy() {
```

Comment:
Looks like there is no testCreateAfterDestroyError

Reply:
No, do you think it's needed? what to check in it?

Reply:
It's not necessary, just it looks like other operation have operation + "Error" tests